src/coreclr/jit/llvm.cpp

```diff
@@ -2727,8 +2878,62 @@ void Llvm::PlaceAndConvertShadowStackLocals()
     for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
     {
         LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+
+        if (varDsc->lvIsParam)
+        {
+            if (_compiler->lvaGetPromotionType(varDsc) == Compiler::PROMOTION_TYPE_INDEPENDENT)
+            {
+                for (unsigned index = 0; index < varDsc->lvFieldCnt; index++)
+                {
+                    unsigned   fieldLclNum = varDsc->lvFieldLclStart + index;
+                    LclVarDsc* fieldVarDsc = _compiler->lvaGetDesc(fieldLclNum);
+                    if (fieldVarDsc->lvRefCnt(RCS_NORMAL) != 0)
+                    {
+                        _compiler->fgEnsureFirstBBisScratch();
+                        LIR::Range& firstBlockRange = LIR::AsRange(_compiler->fgFirstBB);
+
+                        GenTree* fieldValue =
+                            _compiler->gtNewLclFldNode(lclNum, fieldVarDsc->TypeGet(), fieldVarDsc->lvFldOffset);
+
+                        GenTree* fieldStore = _compiler->gtNewStoreLclVar(fieldLclNum, fieldValue);
+                        firstBlockRange.InsertAtBeginning(fieldStore);
+                        firstBlockRange.InsertAtBeginning(fieldValue);
+                    }
+
+                    fieldVarDsc->lvIsStructField = false;
+                    fieldVarDsc->lvParentLcl     = BAD_VAR_NUM;
+                    fieldVarDsc->lvIsParam       = false;
+                }
+
+                varDsc->lvPromoted      = false;
+                varDsc->lvFieldLclStart = BAD_VAR_NUM;
+                varDsc->lvFieldCnt      = 0;
+            }
+            else if (_compiler->lvaGetPromotionType(varDsc) == Compiler::PROMOTION_TYPE_DEPENDENT)
+            {
+                /* dependent promotion, just mark fields as not lvIsParam */
+                for (unsigned index = 0; index < varDsc->lvFieldCnt; index++)
+                {
+                    unsigned   fieldLclNum = varDsc->lvFieldLclStart + index;
+                    LclVarDsc* fieldVarDsc = _compiler->lvaGetDesc(fieldLclNum);
+                    fieldVarDsc->lvIsParam = false;
+                }
+            }
```

Comment:
Is this unmarking needed? I would expect not, since we skip dependent fields in shadow stack building anyway.

Reply:
If I don't do this, then I hit the assert at 
https://github.com/yowl/runtimelab/blob/42e82dafdff63ecc0cd977278c55b37d711a2322/src/coreclr/jit/llvm.cpp#L606
for `S_P_CoreLib_System_Runtime_Intrinsics_Vector64__GetElement<UInt32>` for the var:
```
0x000001d091bd0d38 [V3: TYP_LONG-"field V00._00 (fldOffset=0x0)"]
```